godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/signals.kt

```diff
@@ -6,7 +6,7 @@ abstract class Signal(
     val name: String
 ) {
     protected fun emitSignal(instance: Object, vararg args: Any?) {
-        instance.emitSignal(name, *args)
+        instance.emitSignal(name.camelToSnakeCase(), *args) //not using `camelcaseToUnderscore` to prevent a call to godot for each signal emission
```

Comment:
Instead of doing it every time we want to emit a signal we can do it once in `SignalDelegateProvider`.